homeassistant/components/shelly/__init__.py

```diff
@@ -83,6 +84,16 @@ async def _async_update_data(self):
         if self._unsub_stop is None:
             return None
 
+        if not self.last_update_success:
+            try:
+                async with async_timeout.timeout(10):
+                    self.device = await aioshelly.Device.create(
```

Comment:
This invalidates every reference to blocks that entities hold. This is not good.

Instead we should see what connnection needs to be reinitialized and either set up aiocoap for automatic reconnect or set up aioshelly to be able to reconnect.

Reply:
I see what you mean. Thanks for the hints, I'll give it another try.

Reply:
@balloob Could you help me here a bit? It seems to be UDP6 which fails. Looking at the aiocoap, this is using sockets. The socket is still alive because the server side doesn't call close because it goes down hard. What I see happening (when removing the timeout on update in HA) is that aiocoap is programmed to make 4 retries and then it gets a timeout, but it's not really detected that the connection is broken (the socket send doesn't throw an error).

For aioshelly I can imagine a solution which would put a timeout on `coap_request` and recreates the `coap_context` when timeout happens. It wouldn't invalidate blocks, but is that an acceptable solution?

Reply:
What do you think about changing coap message type? Look here https://github.com/home-assistant/core/issues/40411#issuecomment-702324963

Reply:
@bieniu Seems to work..I'll follow up in that issue.